### PR TITLE
Renamed `incoming_amount` to `incoming_amount_msats`

### PR DESCRIPTION
**Pull request by Mirebella**
I addressed the TODO about renaming `incoming_amount`

### CODE REVIEW

File: gateway/ln-gateway/src/db.rs [hunk 1]
```diff
@@ -431,8 +431,7 @@ struct RegisteredIncomingContractKey(pub PaymentImage);
 pub struct RegisteredIncomingContract {
     pub federation_id: FederationId,
     /// The amount of the incoming contract, in msats.
-    /// TODO: Rename to `incoming_amount_msats` to be more explicit.
-    pub incoming_amount: u64,
```
Comment (elsirion) [inline gateway/ln-gateway/src/db.rs:435]: @joschisan why is this a u64 in the first place and not an `Amount`?

**Comment by elsirion** [comment]
Thx for the contribution, trying to figure out if that's actually the correct move here :)

File: gateway/ln-gateway/src/db.rs (hunk 1 above)
Comment (joschisan) [reply gateway/ln-gateway/src/db.rs:435]: Not sure why exactly here - generally we have some u64 msat amounts in the lnv2 module instead for easy interop with ln invoice amounts. Looks like in this case it should actually be an Amount though.

**Comment by elsirion** [comment]
Looks reasonable to me, just needs a rebase :)

**Review by Kodylow — APPROVED** [review]
rebased

**Comment by bradleystachurski** [comment]
Is this a safe rename? Is it possible to save the old struct in the db, then fail to read/decode the bytes after this change since the field name changed? If so, then this will require a migration.

**Comment by bradleystachurski** [comment]
Capturing an out-of-band sync with @joschisan: this is safe since we'll decode the struct based on the ordering of the fields, not the field names.

**Pull request merged**